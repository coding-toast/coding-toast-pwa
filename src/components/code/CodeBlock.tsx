import * as React from 'react';
import { Prism as SyntaxHighlighter } from 'react-syntax-highlighter';
import { atomDark } from 'react-syntax-highlighter/dist/cjs/styles/prism';

interface ICodeBlockProps {
  language: string;
  value: string;
}

const CodeBlock: React.FC<ICodeBlockProps> = (props) => {
  return (
    <div>
      <SyntaxHighlighter language={props.language} style={atomDark} showLineNumbers>
        {props.value}
      </SyntaxHighlighter>
      <br />
    </div>
  );
};

export default CodeBlock;
