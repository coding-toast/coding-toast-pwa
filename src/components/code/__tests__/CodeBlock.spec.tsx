import * as React from 'react';
import CodeBlock from '../CodeBlock';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

describe('<CodeBlock />', () => {
  it('Should render properly', () => {
    const wrapper = shallow(<CodeBlock language='jsx' value='<pre><code>console.log("Hello World!")' />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
