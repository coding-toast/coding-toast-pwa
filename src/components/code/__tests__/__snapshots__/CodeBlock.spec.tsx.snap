// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CodeBlock /> Should render properly 1`] = `
<div>
  <SyntaxHighlighter
    language="jsx"
    showLineNumbers={true}
    style={
      Object {
        ".language-css .token.string": Object {
          "color": "#87C38A",
        },
        ".namespace": Object {
          "Opacity": ".7",
        },
        ".style .token.string": Object {
          "color": "#87C38A",
        },
        ":not(pre) > code[class*=\\"language-\\"]": Object {
          "background": "#1d1f21",
          "borderRadius": ".3em",
          "padding": ".1em",
        },
        "atrule": Object {
          "color": "#F9EE98",
        },
        "attr-name": Object {
          "color": "#A8FF60",
        },
        "attr-value": Object {
          "color": "#F9EE98",
        },
        "bold": Object {
          "fontWeight": "bold",
        },
        "boolean": Object {
          "color": "#99CC99",
        },
        "builtin": Object {
          "color": "#A8FF60",
        },
        "cdata": Object {
          "color": "#7C7C7C",
        },
        "char": Object {
          "color": "#A8FF60",
        },
        "class-name": Object {
          "color": "#FFFFB6",
          "textDecoration": "underline",
        },
        "code[class*=\\"language-\\"]": Object {
          "MozHyphens": "none",
          "MozTabSize": "4",
          "OTabSize": "4",
          "WebkitHyphens": "none",
          "color": "#c5c8c6",
          "direction": "ltr",
          "fontFamily": "Inconsolata, Monaco, Consolas, 'Courier New', Courier, monospace",
          "hyphens": "none",
          "lineHeight": "1.5",
          "msHyphens": "none",
          "tabSize": "4",
          "textAlign": "left",
          "textShadow": "0 1px rgba(0, 0, 0, 0.3)",
          "whiteSpace": "pre",
          "wordBreak": "normal",
          "wordSpacing": "normal",
        },
        "comment": Object {
          "color": "#7C7C7C",
        },
        "constant": Object {
          "color": "#99CC99",
        },
        "deleted": Object {
          "color": "#f92672",
        },
        "doctype": Object {
          "color": "#7C7C7C",
        },
        "entity": Object {
          "color": "#FFFFB6",
          "cursor": "help",
        },
        "function": Object {
          "color": "#DAD085",
        },
        "important": Object {
          "color": "#fd971f",
          "fontWeight": "bold",
        },
        "inserted": Object {
          "color": "#A8FF60",
        },
        "italic": Object {
          "fontStyle": "italic",
        },
        "keyword": Object {
          "color": "#96CBFE",
        },
        "number": Object {
          "color": "#FF73FD",
        },
        "operator": Object {
          "color": "#EDEDED",
        },
        "pre[class*=\\"language-\\"]": Object {
          "MozHyphens": "none",
          "MozTabSize": "4",
          "OTabSize": "4",
          "WebkitHyphens": "none",
          "background": "#1d1f21",
          "borderRadius": "0.3em",
          "color": "#c5c8c6",
          "direction": "ltr",
          "fontFamily": "Inconsolata, Monaco, Consolas, 'Courier New', Courier, monospace",
          "hyphens": "none",
          "lineHeight": "1.5",
          "margin": ".5em 0",
          "msHyphens": "none",
          "overflow": "auto",
          "padding": "1em",
          "tabSize": "4",
          "textAlign": "left",
          "textShadow": "0 1px rgba(0, 0, 0, 0.3)",
          "whiteSpace": "pre",
          "wordBreak": "normal",
          "wordSpacing": "normal",
        },
        "prolog": Object {
          "color": "#7C7C7C",
        },
        "property": Object {
          "color": "#96CBFE",
        },
        "punctuation": Object {
          "color": "#c5c8c6",
        },
        "regex": Object {
          "color": "#E9C062",
        },
        "selector": Object {
          "color": "#A8FF60",
        },
        "string": Object {
          "color": "#A8FF60",
        },
        "symbol": Object {
          "color": "#f92672",
        },
        "tag": Object {
          "color": "#96CBFE",
        },
        "url": Object {
          "color": "#96CBFE",
        },
        "variable": Object {
          "color": "#C6C5FE",
        },
      }
    }
  >
    &lt;pre&gt;&lt;code&gt;console.log("Hello World!")
  </SyntaxHighlighter>
  <br />
</div>
`;
